api/src/wfl/log.clj

```diff
@@ -0,0 +1,72 @@
+(ns wfl.log
+  "Logging for WFL"
+  (:require [clojure.core]
+            [clojure.data.json :refer [write-str]])
+  (:import [java.sql Timestamp]
+           [java.time OffsetDateTime]))
+
+(defn googleize-field
+  "Prepends the google logging domain to the fields that require it for Google Logging."
+  [field]
+  (let [field-name (name field)]
+    (if (some #(= field-name %) '("labels" "insertId" "operation" "sourceLocation" "spanId" "trace")) (str "logging.googleapis.com/" (name field-name))
+        field-name)))
+
+(defn write-to-stdout
+  "Writes the output to standard output as a json string."
+  [json]
+  (-> (write-str json :key-fn googleize-field :escape-slash false)
+      (.toString)
+      (println)))
+
+(defn get-timestamp
+  "Gets timestamp for the json"
+  []
+  (Timestamp/from (.toInstant (OffsetDateTime/now))))
+
+(defn add-common-fields
+  "Builds the common message before applying type specific fields."
+  [json message]
+  (merge json
+         {:message message
+          :timestamp (get-timestamp)}))
+
+(defn log
+  "Write a log entry to standard output."
+  [severity message & {:as additional-fields}]
+  (let [severity-upper (.toUpperCase severity)]
+    (if (not (some #(= severity-upper %) '("DEFAULT" "DEBUG" "INFO" "NOTICE" "WARNING" "ERROR" "CRITICAL" "ALERT" "EMERGENCY")))
+      (throw (ex-info "Invalid severity provided to logging function." {:severity severity-upper}))
```

Comment:
Not sure this is really necessary - I'd let clients pass in the log level and just convert to the required format - you can use `to-upper` instead of the java method to make it seem more idiomatic, for example:
```clojure 
(-> severity name to-upper)
```